.github/workflows/publish_pypi.yml

```diff
@@ -35,7 +35,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
-          python-version: 3.8
+          python-version: 3.11
```

Comment:
I think we can remove `actions/setup-python` from all of the wheel jobs and use the [`cibuildwheel` action](https://github.com/pypa/cibuildwheel/blob/main/action.yml) instead which does it for us already (also `cibuildwheel` installs its own Python interpreters). Please feel free to do it here or in a separate PR, it's mostly cleanup though

Reply:
Yeah lets make that one a different PR